src/libstd/coverage.rs

```diff
@@ -0,0 +1,83 @@
+//! Code coverage counters and report
```

Comment:
Why is this being added to the standard library?

Reply:
I need to add the library of rustc-injected coverage APIs to a reserved namespace available only to rustc. "core" doesn't make sense to me. I'm not sure if there are any alternatives?

Reply:
One approach is a dedicated crate that is distributed along with the rustc and injected as a dependency when requested during compilation. For example, `-Zprofile` injects `profiler_builtins` crate, which you might potentially also find useful here.

Reply:
I actually started with "coverage_builtins" following that model, but wasn't sure if/how that would make it into the program's library dependencies, and since "coverage_builtins" is not a truly reserved namespace, it made me uneasy.

builtins::coverage would be great, perhaps, if the Rust community is open to reserving the "builtins" namespace for instrumentation APIs. (Or is it already?)

WDYT?

Reply:
I suggest using LLVM's runtime support in compiler_rt instead (along with the llvm.instrprof.increment intrinsic).